File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-1.cpp
class Solution {
public:
    int goodNodes(TreeNode* root) {
        if (!root) return 0;

        stack<pair<TreeNode*, int>> st;
        st.push({root, root->val});
        int res = 0;

        while (!st.empty()) {
            auto [node, maxSoFar] = st.top();
            st.pop();

            if (node->val >= maxSoFar) res++;
            int newMax = max(maxSoFar, node->val);

            if (node->right) st.push({node->right, newMax});
            if (node->left) st.push({node->left, newMax});
        }

        return res;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        res=0
        s=[]
        prev =prices[0]
        for i in range(1,len(prices)):
            if prices[i]>prev:
                res= max( prices[i]-prev,res)
            else :
                prev=prices[i]
        return res